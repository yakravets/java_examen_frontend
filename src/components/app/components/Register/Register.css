.content_register{
    align-items: center;
    padding: 10px;
}

.block_data_register{
    border: 1px #1890ff solid;
    border-radius: 10px;
    padding-right: 20px;
    margin-top: 10px;
    margin-bottom: 10px;
}