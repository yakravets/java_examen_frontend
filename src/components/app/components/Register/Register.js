import React from 'react';
import { Form, Input, Select, Checkbox, Button, Layout, Divider, message } from 'antd';
import './Register.css';
import config from '../../config.js';
import axios from 'axios';

const registerUrl = config.api_url + config.register_register;

const formItemLayout = {
  labelCol: {
    xs: { span: 16 },
    sm: { span: 8 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 24 },
  },
};
const tailFormItemLayout = {
  wrapperCol: {
    xs: {
      span: 12,
      offset: 0,
    },
    sm: {
      span: 16,
      offset: 4,
    },
  },
};

const RegistrationForm = () => {
  const [form] = Form.useForm();

  const onFinish = (values) => {
    
    axios.post(
        registerUrl,
        JSON.stringify(values),
        {
            headers: {
                'Content-Type': 'application/json'
            }
        })
    .then(res => {
        
        if(res.data.success){
            window.location.href='/login';
        }
        else{
            message.error(res.data.message);
        }
      })
    .catch(function (error) {
        message.error("Network error. Try again await.");
      });
    
      console.log('Received values of form: ', values);

  };

  return (
      <>
        <Layout className='content_register'>            
            <Form
                {...formItemLayout}
                form={form}
                name="register"
                onFinish={onFinish}
                scrollToFirstError
            >
                <Layout orientation='vertical' className="block_data_register">
                    <Divider orientation="left">Access:</Divider>
                    <Form.Item
                        name="email"
                        label="E-mail"
                        rules={[
                        {
                            type: 'email',
                            message: 'The input is not valid E-mail!',
                        },
                        {
                            required: true,
                            message: 'Please input your E-mail!',
                        },
                        ]}
                    >
                        <Input />
                    </Form.Item>
                    
                    <Form.Item
                        name="password"
                        label="Password"
                        rules={[
                        {
                            required: true,
                            message: 'Please input your password!',
                        },
                        ]}
                        hasFeedback
                    >
                        <Input.Password />
                    </Form.Item>

                    <Form.Item
                        name="confirm"
                        label="Confirm"
                        dependencies={['password']}
                        hasFeedback
                        rules={[
                        {
                            required: true,
                            message: 'Please confirm your password!',
                        },
                        ({ getFieldValue }) => ({
                            validator(_, value) {
                            if (!value || getFieldValue('password') === value) {
                                return Promise.resolve();
                            }
                            return Promise.reject(new Error('The two passwords that you entered do not match!'));
                            },
                        }),
                        ]}
                    >
                        <Input.Password />
                    </Form.Item>
                </Layout>
                
                <Layout orientation='vertical' className="block_data_register">                    
                    <Divider orientation="left">Contact information data:</Divider>
                    <Form.Item
                        name="firstName"
                        label="First name"
                        rules={[{ required: true, message: 'Please input your First name!', whitespace: true }]}
                    >
                        <Input />
                    </Form.Item>
                    
                    <Form.Item
                        name="lastName"
                        label="Last name"
                        rules={[{ required: true, message: 'Please input your Last name!', whitespace: true }]}
                    >
                        <Input />
                    </Form.Item>

                    <Form.Item
                        name="gender"
                        label="Gender"
                        rules={[{ required: true, message: 'Please select gender!' }]}
                    >
                        <Select placeholder="select your gender">
                        <Select.Option value="Male">Male</Select.Option>
                        <Select.Option value="Female">Female</Select.Option>
                        </Select>
                    </Form.Item>

                    <Form.Item
                        name="agreement"
                        valuePropName="checked"
                        rules={[
                        {
                            validator: (_, value) =>
                            value ? Promise.resolve() : Promise.reject(new Error('Should accept agreement')),
                        },
                        ]}
                        {...tailFormItemLayout}
                    >
                        <Checkbox>
                            I have read the <a href="">agreement</a>
                        </Checkbox>
                    </Form.Item>
                    <Form.Item {...tailFormItemLayout}>
                        <Button type="primary" htmlType="submit">
                            Register
                        </Button>
                    </Form.Item>
                </Layout>
            </Form>
        </Layout>
    </>
  );
};

export default RegistrationForm;