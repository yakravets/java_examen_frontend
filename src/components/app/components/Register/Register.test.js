import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
import toJson from "enzyme-to-json";

import Register from './Register';

it("renders correctly", () => {
    const tree = shallow(<Register />);
    expect(toJson(tree)).toMatchSnapshot();
});

it("renders without crashing", () => {
    shallow(<Register />);
});