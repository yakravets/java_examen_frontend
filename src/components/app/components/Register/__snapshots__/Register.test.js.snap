// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <Layout
    className="content_register"
  >
    <ForwardRef(InternalForm)
      form={
        Object {
          "__INTERNAL__": Object {
            "itemRef": [Function],
          },
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "getInternalHooks": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "resetFields": [Function],
          "scrollToField": [Function],
          "setFields": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
        }
      }
      labelCol={
        Object {
          "sm": Object {
            "span": 8,
          },
          "xs": Object {
            "span": 16,
          },
        }
      }
      name="register"
      onFinish={[Function]}
      scrollToFirstError={true}
      wrapperCol={
        Object {
          "sm": Object {
            "span": 24,
          },
          "xs": Object {
            "span": 24,
          },
        }
      }
    >
      <Layout
        className="block_data_register"
        orientation="vertical"
      >
        <Divider
          orientation="left"
        >
          Access:
        </Divider>
        <FormItem
          label="E-mail"
          name="email"
          rules={
            Array [
              Object {
                "message": "The input is not valid E-mail!",
                "type": "email",
              },
              Object {
                "message": "Please input your E-mail!",
                "required": true,
              },
            ]
          }
        >
          <Input
            type="text"
          />
        </FormItem>
        <FormItem
          hasFeedback={true}
          label="Password"
          name="password"
          rules={
            Array [
              Object {
                "message": "Please input your password!",
                "required": true,
              },
            ]
          }
        >
          <Password
            action="click"
            iconRender={[Function]}
            visibilityToggle={true}
          />
        </FormItem>
        <FormItem
          dependencies={
            Array [
              "password",
            ]
          }
          hasFeedback={true}
          label="Confirm"
          name="confirm"
          rules={
            Array [
              Object {
                "message": "Please confirm your password!",
                "required": true,
              },
              [Function],
            ]
          }
        >
          <Password
            action="click"
            iconRender={[Function]}
            visibilityToggle={true}
          />
        </FormItem>
      </Layout>
      <Layout
        className="block_data_register"
        orientation="vertical"
      >
        <Divider
          orientation="left"
        >
          Contact information data:
        </Divider>
        <FormItem
          label="First name"
          name="firstName"
          rules={
            Array [
              Object {
                "message": "Please input your First name!",
                "required": true,
                "whitespace": true,
              },
            ]
          }
        >
          <Input
            type="text"
          />
        </FormItem>
        <FormItem
          label="Last name"
          name="lastName"
          rules={
            Array [
              Object {
                "message": "Please input your Last name!",
                "required": true,
                "whitespace": true,
              },
            ]
          }
        >
          <Input
            type="text"
          />
        </FormItem>
        <FormItem
          label="Gender"
          name="gender"
          rules={
            Array [
              Object {
                "message": "Please select gender!",
                "required": true,
              },
            ]
          }
        >
          <ForwardRef(InternalSelect)
            placeholder="select your gender"
          >
            <Option
              value="Male"
            >
              Male
            </Option>
            <Option
              value="Female"
            >
              Female
            </Option>
          </ForwardRef(InternalSelect)>
        </FormItem>
        <FormItem
          name="agreement"
          rules={
            Array [
              Object {
                "validator": [Function],
              },
            ]
          }
          valuePropName="checked"
          wrapperCol={
            Object {
              "sm": Object {
                "offset": 4,
                "span": 16,
              },
              "xs": Object {
                "offset": 0,
                "span": 12,
              },
            }
          }
        >
          <Checkbox>
            I have read the 
            <a
              href=""
            >
              agreement
            </a>
          </Checkbox>
        </FormItem>
        <FormItem
          wrapperCol={
            Object {
              "sm": Object {
                "offset": 4,
                "span": 16,
              },
              "xs": Object {
                "offset": 0,
                "span": 12,
              },
            }
          }
        >
          <Button
            htmlType="submit"
            type="primary"
          >
            Register
          </Button>
        </FormItem>
      </Layout>
    </ForwardRef(InternalForm)>
  </Layout>
</Fragment>
`;
